<a href="https://colab.research.google.com/github/Shivachikkonda02/Python-Project/blob/main/Virat%20Kohli%20Performance%20Analysis%20.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

import plotly.express as px
import plotly.graph_objects as go

In [23]:
df=pd.read_csv("/content/Virat Kohli.csv")
df

Runs   BF  4s  6s      SR  Pos Dismissal  Inns     Opposition  \
0      12   22   1   0   54.54  2.0       lbw     1    v Sri Lanka   
1      37   67   6   0   55.22  2.0    caught     2    v Sri Lanka   
2      25   38   4   0   65.78  1.0   run out     1    v Sri Lanka   
3      54   66   7   0   81.81  1.0    bowled     1    v Sri Lanka   
4      31   46   3   1   67.39  1.0       lbw     2    v Sri Lanka   
..    ...  ...  ..  ..     ...  ...       ...   ...            ...   
127    45   51   2   1   88.23  3.0    caught     2  v New Zealand   
128    65   76   2   1   85.52  3.0    caught     1  v New Zealand   
129   122  105   8   5  116.19  3.0    caught     2      v England   
130     8    5   2   0  160.00  3.0    caught     1      v England   
131    55   63   8   0   87.30  3.0    caught     2      v England   

            Ground Start Date  
0         Dambulla  18-Aug-08  
1         Dambulla  20-Aug-08  
2    Colombo (RPS)  24-Aug-08  
3    Colombo (RPS)  27-Aug-08  
4    Colombo (RPS)  29-Aug-08  
..             ...        ...  
127         Ranchi  26-Oct-16  
128  Visakhapatnam  29-Oct-16  
129           Pune  15-Jan-17  
130        Cuttack  19-Jan-17  
131        Kolkata  22-Jan-17  

[132 rows x 11 columns]

In [24]:
df.head()

Runs  BF  4s  6s     SR  Pos Dismissal  Inns   Opposition         Ground  \
0    12  22   1   0  54.54  2.0       lbw     1  v Sri Lanka       Dambulla   
1    37  67   6   0  55.22  2.0    caught     2  v Sri Lanka       Dambulla   
2    25  38   4   0  65.78  1.0   run out     1  v Sri Lanka  Colombo (RPS)   
3    54  66   7   0  81.81  1.0    bowled     1  v Sri Lanka  Colombo (RPS)   
4    31  46   3   1  67.39  1.0       lbw     2  v Sri Lanka  Colombo (RPS)   

  Start Date  
0  18-Aug-08  
1  20-Aug-08  
2  24-Aug-08  
3  27-Aug-08  
4  29-Aug-08

In [25]:
df.tail()

Runs   BF  4s  6s      SR  Pos Dismissal  Inns     Opposition  \
127    45   51   2   1   88.23  3.0    caught     2  v New Zealand   
128    65   76   2   1   85.52  3.0    caught     1  v New Zealand   
129   122  105   8   5  116.19  3.0    caught     2      v England   
130     8    5   2   0  160.00  3.0    caught     1      v England   
131    55   63   8   0   87.30  3.0    caught     2      v England   

            Ground Start Date  
127         Ranchi  26-Oct-16  
128  Visakhapatnam  29-Oct-16  
129           Pune  15-Jan-17  
130        Cuttack  19-Jan-17  
131        Kolkata  22-Jan-17

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132 entries, 0 to 131
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Runs        132 non-null    int64  
 1   BF          132 non-null    int64  
 2   4s          132 non-null    int64  
 3   6s          132 non-null    int64  
 4   SR          132 non-null    float64
 5   Pos         132 non-null    float64
 6   Dismissal   132 non-null    object 
 7   Inns        132 non-null    int64  
 8   Opposition  132 non-null    object 
 9   Ground      132 non-null    object 
 10  Start Date  132 non-null    object 
dtypes: float64(2), int64(5), object(4)
memory usage: 11.5+ KB


In [27]:
df.describe()

Runs          BF          4s          6s          SR         Pos  \
count  132.000000  132.000000  132.000000  132.000000  132.000000  132.000000   
mean    46.848485   50.871212    4.371212    0.545455   76.991212    3.303030   
std     41.994635   38.729716    4.404032    1.086795   37.874491    0.873174   
min      0.000000    0.000000    0.000000    0.000000    0.000000    1.000000   
25%     10.000000   17.750000    1.000000    0.000000   61.702500    3.000000   
50%     32.500000   42.500000    3.000000    0.000000   78.710000    3.000000   
75%     80.250000   82.250000    7.000000    1.000000  100.000000    4.000000   
max    154.000000  140.000000   18.000000    7.000000  209.090000    7.000000   

             Inns  
count  132.000000  
mean     1.575758  
std      0.496110  
min      1.000000  
25%      1.000000  
50%      2.000000  
75%      2.000000  
max      2.000000

In [28]:
df.dtypes

Runs            int64
BF              int64
4s              int64
6s              int64
SR            float64
Pos           float64
Dismissal      object
Inns            int64
Opposition     object
Ground         object
Start Date     object
dtype: object

In [29]:
# Checking Null Values
df.isnull().sum()


Runs          0
BF            0
4s            0
6s            0
SR            0
Pos           0
Dismissal     0
Inns          0
Opposition    0
Ground        0
Start Date    0
dtype: int64

In [30]:
# Total Runs Between 18-Aug-08 - 22-Jan-17

df['Runs'].sum()

6184

In [31]:
# Average Runs Between 18-Aug-08 - 22-Jan-17
df['Runs'].mean()

46.84848484848485

In [34]:
matches=df.index
figure=px.line(df,x=matches,y='Runs',
               title='Runs Scored by Virat Kohli Between 18-Aug-08 - 22-Jan-17')
figure.show()

In [35]:
df.Pos

0      2.0
1      2.0
2      1.0
3      1.0
4      1.0
      ... 
127    3.0
128    3.0
129    3.0
130    3.0
131    3.0
Name: Pos, Length: 132, dtype: float64

In [36]:
df['Pos'].value_counts()

Pos
3.0    91
4.0    28
2.0     3
1.0     3
7.0     3
5.0     3
6.0     1
Name: count, dtype: int64

In [38]:
# Batting Positions
df['Pos']=df['Pos'].map({3.0: "Batting At 3", 4.0: "Batting At 4", 2.0: "Batting At 2",
                               1.0: "Batting At 1", 7.0:"Batting At 7", 5.0:"Batting At 5",
                               6.0: "batting At 6"})

In [39]:
df['Pos'].value_counts()

Pos
Batting At 3    91
Batting At 4    28
Batting At 2     3
Batting At 1     3
Batting At 7     3
Batting At 5     3
batting At 6     1
Name: count, dtype: int64

In [43]:
Pos=df['Pos'].value_counts()

In [44]:
Pos.index

Index(['Batting At 3', 'Batting At 4', 'Batting At 2', 'Batting At 1',
       'Batting At 7', 'Batting At 5', 'batting At 6'],
      dtype='object', name='Pos')

In [45]:
Pos.values

array([91, 28,  3,  3,  3,  3,  1])

In [46]:

label = Pos.index
counts = Pos.values
colors = ['gold','lightgreen', "pink", "blue", "skyblue", "cyan", "orange"] #Specified colors

fig = go.Figure(data=[go.Pie(labels=label, values=counts)])
fig.update_layout(title_text='Number of Matches At Different Batting Positions') #Title provided
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=30,
                  marker=dict(colors=colors, line=dict(color='black', width=3))) #Added the feature that shows percentage when we hover over the chart
fig.show()


In [47]:
centuries = df.query("Runs >= 100")
figure = px.bar(centuries, x=centuries["Inns"], y = centuries["Runs"],
                color = centuries["Runs"],
                title="Centuries By Virat Kohli in First Innings Vs. Second Innings")
figure.show()

In [49]:
dismissal = df["Dismissal"].value_counts()
label = dismissal.index
counts = dismissal.values
colors = ['gold','lightgreen', "pink", "blue", "skyblue", "cyan", "orange"]

fig = go.Figure(data=[go.Pie(labels=label, values=counts)])
fig.update_layout(title_text='Dismissals of Virat Kohli')
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=30,
                  marker=dict(colors=colors, line=dict(color='black', width=3)))
fig.show()

In [51]:
figure=px.bar(df,x=df['Opposition'], y = df['Runs'] , color=df['Runs'],
              title='Most Runs Againist Teams')

figure.show()

In [52]:
figure = px.bar(centuries, x=centuries["Opposition"], y = centuries["Runs"],
                color = centuries["Runs"],
                title="Most Centuries Against Teams")
figure.show()

In [53]:
strike_rate=df.query('SR>=120')
strike_rate

Runs  BF  4s  6s      SR           Pos Dismissal  Inns     Opposition  \
8      27  19   4   0  142.10  Batting At 7    bowled     1    v Sri Lanka   
32    100  83   8   2  120.48  Batting At 4   not out     1   v Bangladesh   
56     23  11   3   0  209.09  batting At 6   not out     1  v West Indies   
76     43  34   4   1  126.47  Batting At 3    caught     1      v England   
78    102  83  13   2  122.89  Batting At 3    caught     1  v West Indies   
83    100  52   8   7  192.30  Batting At 3   not out     2    v Australia   
85    115  66  18   1  174.24  Batting At 3   not out     2    v Australia   
93     78  65   7   2  120.00  Batting At 3    caught     2  v New Zealand   
130     8   5   2   0  160.00  Batting At 3    caught     1      v England   

            Ground Start Date  
8           Rajkot  15-Dec-09  
32           Dhaka  19-Feb-11  
56          Indore   8-Dec-11  
76      Birmingham  23-Jun-13  
78   Port of Spain   5-Jul-13  
83          Jaipur  16-Oct-13  
85          Nagpur  30-Oct-13  
93        Hamilton  22-Jan-14  
130        Cuttack  19-Jan-17

In [54]:
figure = px.bar(strike_rate, x = strike_rate["Inns"],
                y = strike_rate["SR"],
                color = strike_rate["SR"],
            title="Virat Kohli's High Strike Rates in First Innings Vs. Second Innings")
figure.show()

In [55]:
figure = px.scatter(data_frame = df, x="Runs",
                    y="4s", size="SR", trendline="ols",
                    title="Relationship Between Runs Scored and Fours")
figure.show()


In [56]:
figure = px.scatter(data_frame = df, x="Runs",
                    y="6s", size="SR", trendline="ols",
                    title= "Relationship Between Runs Scored and Sixes")
figure.show()


Conclusions

1. Total Runs : 6184
2. Average runs : 46.84848484848485
3. 68% of all the innings played by Virat Kohli, he batted in the third position.
4. 72% of the total runs scored by Virat Kohli are while batting at 3rd position.
5. Most of the centuries are scored while batting in the second innings.
6. 63.6 Virat Kohli gets out by getting caught by the fielder or the keeper.
7. Most of the centuries scored by Virat Kohli were against Australia.
